### ISSUE TITLE
equation \label and \ref

### ISSUE THREAD

**Issue by slymz** [mathjax/MathJax#71]
Feature request!  Can we have \label and \ref facilities for equations?

This may be a popular one indeed, but I couldn't find it on issue list (or mention of it in the documentation). And is basically been discussed long back: http://sourceforge.net/projects/mathjax/forums/forum/948700/topic/3817124

Thank you!

**Comment by simonmysun** [comment]
Hi, 

Thank you all for the amazing project and this amazing feature. Here I am have a small problem. When modifying maths with `\label` inside, the labels will be multiply defined(to reproduce this you may try to edit a formula with `\label` in any MathJax live demo). 

I believe it can be fixed by removing the label definition in MathJax when replacing the HTML elements. But I haven't find any interface to inform this removing, or any code that can clear the definitions without refreshing the page. 

Does any have any idea what I can do? 

Thank you in advance.

**Comment by pkra** [comment]
Please don't re-use threads. 

You'll want to [read the documentation](http://docs.mathjax.org/en/latest/advanced/typeset.html#reset-equation-numbers) on resetting equation numbers.

**Comment by simonmysun** [comment]
Alright I thought it is an incomplete feature rather than bug. I'll start a new issue. 

The equation numbers can be resetted, but the labels cannot.

**Comment by dpvc** [comment]
The labels are reset as well by `resetEquationNumbers()`, though the documentation could be clearer about that.

**Comment by simonmysun** [comment]
Thank you very much! It must be my wrong usage leading to the failure. Thank you again. Sorry for disturbing.

**Comment by cben** [comment]
It seems `MathJax.InputJax.TeX.resetEquationNumbers()` was a 2.x construct: http://docs.mathjax.org/en/v2.7-latest/advanced/typeset.html#reset-automatic-equation-numbering
optional args undocumented but explained by Davide at https://github.com/mathjax/MathJax/issues/1705#issuecomment-282743204, thanks.
In 3.x replaced by `MathJax.texReset()` http://docs.mathjax.org/en/v3.0-latest/web/typeset.html#tex-reset
[of course, tons of things changed in 3.0.  this is mostly to help future me find this again as I'm still on 2.x ;-)  It's a bit unfortunate lots of 2.x doc links are now breaking if they were to "latest/"...]
